src/payment/strategies/square/square-payment-strategy.ts

```diff
@@ -103,27 +130,108 @@ export default class SquarePaymentStrategy extends PaymentStrategy {
                         this._paymentForm.setPostalCode(billingAddress.postalCode);
                     }
                 },
+
                 unsupportedBrowserDetected: () => {
                     deferred.reject(new UnsupportedBrowserError());
                 },
-                cardNonceResponseReceived: (errors, nonce) => {
-                    this._cardNonceResponseReceived(errors, nonce);
+
+                cardNonceResponseReceived: (errors?, nonce?, cardData?, billingContact?, shippingContact?) => {
+
+                    if (cardData && cardData.digital_wallet_type !== DigitalWalletType.none) {
+                        this._setExternalCheckoutData(nonce, cardData, billingContact, shippingContact)
+                            .then(() => this._paymentInstrumentSelected(methodId))
+                            .then(() => squareOptions.onPaymentSelect && squareOptions.onPaymentSelect());
+                    } else {
+                        this._cardNonceResponseReceived(nonce, errors);
+                    }
+                },
+
+                methodsSupported: (methods: any) => {
+                    const masterpassBtn = document.getElementById('sq-masterpass');
+                    if (masterpassBtn !== null) {
+                        if (methods.masterpass === true) {
+                            masterpassBtn.style.display = 'inline-block';
+                        }
+                    }
+                },
+
+                /*
+                 * callback function: createPaymentRequest
+                 * Triggered when: a digital wallet payment button is clicked.
+                */
+                createPaymentRequest: () => {
+                    const state = this._store.getState();
+                    const checkout = state.checkout.getCheckout();
+                    const storeConfig = state.config.getStoreConfig();
+
+                    if (!checkout) {
+                        throw new MissingDataError(MissingDataErrorType.MissingCheckout);
+                    }
+
+                    if (!storeConfig) {
+                        throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+                    }
+
+                    return {
+                        requestShippingAddress: true,
+                        requestBillingInfo: true,
+                        currencyCode: storeConfig.currency.code,
+                        countryCode: 'US',
```

Comment:
@davidchin I believe it's because Masterpass is US only and this specific method is only triggered when the Masterpass button on the Square payment form is clicked.

Reply:
But shouldn't we always pass real information rather than some hardcoded value? And this callback `createPaymentRequest` applies to all wallet providers (i.e.: ApplePay and GooglePay) not just Masterpass. I know we don't support them now but this might come back to us in the future.

Reply:
@davidchin @Sure-Round Could it be possible if we add [iso-3166-1 package](https://www.npmjs.com/package/iso-3166-1) to be consumed so we avoid coding multiple (semi) implementations of it?